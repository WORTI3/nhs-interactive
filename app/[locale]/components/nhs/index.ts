export { default as Button } from './Button';
export { default as Input } from './Input';
export { Header } from './Header';
export { Footer } from './Footer';
export { wrapper as Wrapper } from './wrapper';
export * from './WarningCallout';