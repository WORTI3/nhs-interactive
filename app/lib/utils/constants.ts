export const LOCAL_KEY_ID = "interactive-local-key";
export const UNRELATED_ANSWER = "not-answerable";
export const LOCAL_DB_URL = "http://127.0.0.1:8090";

export enum OpenAIModel {
  DAVINCI_TURBO = "gpt-3.5-turbo",
};