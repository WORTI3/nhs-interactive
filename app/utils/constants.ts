export const LOCAL_KEY_ID = "interactive-local-key";
export const UNRELATED_ANSWER = "not-answerable";
export const LOCAL_DB_URL = "http://127.0.0.1:8090";
export const SERVICE_NAME = "NHS Interactive";
export const SERVICE_DESC = "AI powered search.";
export const SERVICE_URL = "https://nhs-interactive.vercel.app";
export const LOCALES = ['en', 'cy', 'gd'];
export const USE_AI_RESPONSE_KEY = "use-ai-response";

export enum OpenAIModel {
  DAVINCI_TURBO = "gpt-3.5-turbo",
};