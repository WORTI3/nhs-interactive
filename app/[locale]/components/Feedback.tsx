import { Dialog, DialogTitle } from '@headlessui/react';
import { useState } from 'react';
import { Button } from './nhs';
import { useTranslations } from 'next-intl';

const FeedbackDialog = ({ isOpen, onClose, onSubmit }: {isOpen: any, onClose: any, onSubmit: any}) => {
  const translate = useTranslations('search');
  const [comment, setComment] = useState('');
  const [helpful, setHelpful] = useState(false);
  const [error, setError] = useState<string>("");

  const handleSubmit = () => {
    onSubmit({ comment, helpful });
    onClose();
  };

  return (
    <Dialog open={isOpen} onClose={onClose} className="fixed inset-0 overflow-hidden">
      <div className="absolute inset-0 overflow-hidden">
        <div className="absolute inset-0 bg-black opacity-50" />
        <div className="fixed inset-y-0 right-0 max-w-full flex">
          <div className="w-screen max-w-md">
            <div className="h-full bg-white p-4 shadow-xl">
              <DialogTitle>Provide Feedback</DialogTitle>
              <textarea
                className="w-full border rounded mt-4 p-2"
                placeholder="Leave a note..."
                value={comment}
                onChange={(e) => setComment(e.target.value)}
              />
              <div className="mt-4">
                <label>
                  <input
                    type="checkbox"
                    checked={helpful}
                    onChange={(e) => setHelpful(e.target.checked)}
                  />
                  Was this helpful?
                </label>
              </div>
              <div className="mt-4">
                <button
                  className="bg-blue-500 text-white p-2 rounded"
                  onClick={handleSubmit}
                >
                  Submit
                </button>
                <Button
                  id='submit'
                  as='a'
                  data-prevent-double-click
                  onClick={handleSubmit}
                  disabled={!!error && comment === ''}
                >
                  {translate('button.submit')}
                </Button>
                <button
                  className="ml-2 p-2"
                  onClick={onClose}
                >
                  Cancel
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </Dialog>
  );
};

export default FeedbackDialog;
