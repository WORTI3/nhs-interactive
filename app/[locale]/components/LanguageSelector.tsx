import { Fragment } from 'react'
import { Menu, Transition } from '@headlessui/react'
import { IconWorld } from '@tabler/icons-react';
import Link from "next-intl/link";
import { LOCALES } from '../../utils/constants';

function classNames(...classes: any[]) {
  return classes.filter(Boolean).join(' ')
}

export default function LanguageSelector({translate, locale}: {translate: any, locale: string}) {
  return (
    <Menu as='div' className='relative max-lg:absolute max-lg:right-4 float-right inline-block text-left'>
      <div>
        {/* Hover style made need changed to match NHS */}
        <Menu.Button id='language-button' className='inline-flex w-full justify-center gap-x-1.5 rounded-md bg-white px-3 py-2 text-sm font-semibold text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300 hover:bg-gray-50'>
          {translate('button.language')}
          <IconWorld
            className='-mr-1 h-5 w-5 text-gray-400'
            aria-hidden='true'
          />
        </Menu.Button>
      </div>

      <Transition
        as={Fragment}
        enter='transition ease-out duration-100'
        enterFrom='transform opacity-0 scale-95'
        enterTo='transform opacity-100 scale-100'
        leave='transition ease-in duration-75'
        leaveFrom='transform opacity-100 scale-100'
        leaveTo='transform opacity-0 scale-95'
      >
        <Menu.Items className='absolute right-0 z-10 mt-2 w-56 origin-top-right rounded-md bg-white shadow-lg ring-1 ring-black ring-opacity-5 focus:outline-none'>
          <div className='langs'>
            {LOCALES.map(lang => {
              if (lang !== locale) {
                return (
                  <Menu.Item key={lang} data-testid="language-item">
                    {({ active }) => (
                      <Link
                        href='/'
                        locale={lang}
                        className={classNames(
                          active ? "bg-gray-100 text-gray-900" : "text-gray-700",
                          "block px-4 py-2 text-sm rounded-md"
                        )}
                      >
                        <p className='mb-0 py-2'>
                          <span
                            className={`mr-4 h-5 w-5 rounded-md fi fis fi-${lang === 'en' ? 'gb-eng' : lang === 'cy' ? 'gb-wls' : 'gb-sct'}`}
                            aria-hidden='true'
                          />
                          {lang === 'en' ? 'English' : lang === 'cy' ? 'Cymraeg' : 'Gaelic'}
                        </p>
                      </Link>
                    )}
                  </Menu.Item>
                );
              }
              return null;
            })}
          </div>
        </Menu.Items>
      </Transition>
    </Menu>
  );
}
