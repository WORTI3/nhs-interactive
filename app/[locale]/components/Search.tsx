"use client";

import { FC, KeyboardEvent, useState } from "react";
import { useTranslations } from "next-intl";
import { z } from "zod";
import { Button, Input } from "./nhs";
import { ALLOWED_SEARCH_CHARS_REGEX, SearchProps, fetchAnswer, validateInput } from "../../utils";
import { parseAsString, useQueryState } from 'next-usequerystate';
import { SkeletonResult } from "./SkeletonResult";

const querySchema = z.string().min(4).max(100).regex(ALLOWED_SEARCH_CHARS_REGEX);

export const Search: FC<SearchProps> = ({
  onSearch,
  onResultUpdate: onAnswerUpdate,
  onDone,
  setResultIdStore,
}) => {
  const translate = useTranslations('search');
  const [query] = useQueryState('q', parseAsString.withDefault(""));
  const [loading, setLoading] = useState<boolean>(false);
  const [error, setError] = useState<string>("");
  
  const handleKeyDown = (e: KeyboardEvent<HTMLTextAreaElement>) => {
    if (e.key === "Enter") {
      handleSearch();
    }
  };

  const handleSearch = async () => {
    const sanitisedQuery = validateInput(query);
    const result = querySchema.safeParse(sanitisedQuery);
    if (!result.success) {
      // dont care for multiple errors here so using the first in array is fine.
      setError(translate(`errors.${result.error.issues[0].code}`));
      return;
    }

    setLoading(true);
    try {
      await fetchAnswer(query, onAnswerUpdate, onSearch, onDone, translate('lang'), setResultIdStore);
    } catch (error) {
      setError('Service currently unavailable. Please try again later.');
    }
    setLoading(false);
  };

  return (
    <>
      {loading ? (
        <SkeletonResult />
      ) : (
        <div className='relative w-full'>
          <Input
            id="search-input"
            label={translate('query')}
            name="search"
            hint={translate('hint')}
            placeholder={translate('placeholder')}
            value={query}
            onKeyDown={handleKeyDown}
            error={error}
          />
          <Button
            id='submit'
            data-prevent-double-click
            onClick={handleSearch}
            disabled={!!error && query === ''}
          >
            {translate('button.submit')}
          </Button>
          <span className="nhsuk-body-s">{translate('warning.content')}</span>
        </div>
      )}
    </>
  );
};
