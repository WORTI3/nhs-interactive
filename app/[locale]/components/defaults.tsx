import { OpenAIModel } from "../../utils/constants";

/**
 * `export const ModelInfo` is defining a React functional component that returns a paragraph element displaying
 * information about a specific OpenAI model. The model information is accessed through the `OpenAIModel` constant defined
 * in the `constants.ts` file. This component can be imported and used in other parts of the codebase.
 * 
 * @function
 * @name ModelInfo
 * @kind variable
 * @returns {JSX.Element}
 * @exports
 */
export const ModelInfo = () => {
  return (
    <p className="nhsuk-body-s nhsuk-u-secondary-text-color nhsuk-u-margin-top-7 nhsuk-u-margin-bottom-0">
      Model: { OpenAIModel.DAVINCI_TURBO }
    </p>
  );
}

/**
 * This code exports a function named `makeSourcesLinks` that takes in two parameters: `answer` and `sourceLinks`. The
 * function splits the `answer` string using a regular expression and maps over the resulting array. If a part of the
 * string matches the regular expression `/\[[0-9]+\]/`, it extracts the number within the square brackets, uses it to
 * retrieve a link from the `sourceLinks` array, and returns an anchor tag with the link and the original part of the
 * string. If the part of the string does not match the regular expression, it returns the original part of the string. The
 * function then returns an array of elements that can be rendered in the UI.
 * 
 * @function
 * @name makeSourcesLinks
 * @kind variable
 * @exports
 */
export const makeSourcesLinks = (answer: string, sourceLinks: string[]) => {
  const elements = answer.split(/(\[[0-9]+\])/).map((part, index) => {
    if (/\[[0-9]+\]/.test(part)) {
      const link = sourceLinks[parseInt(part.replace(/[\[\]]/g, "")) - 1];

      return (
        <a
          key={index}
          className="hover:cursor-pointer text-nhs-blue text-base"
          href={link}
          target="_blank"
          rel="noopener noreferrer"
        >
          {part}
        </a>
      );
    } else {
      return part;
    }
  });

  return elements;
};