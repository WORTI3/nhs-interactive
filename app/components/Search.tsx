"use client";

import { ChangeEvent, FC, KeyboardEvent, useEffect, useRef, useState } from "react";
import { ActionLink, Button, Input, InsetText } from "nhsuk-react-components";
import { Source } from "@/types";
import endent from "endent";
import { LOCAL_KEY_ID, UNRELATED_ANSWER } from "../lib/utils/constants";
import { SearchProps } from "../lib/utils/interfaces";

export const Search: FC<SearchProps> = ({
  onSearch,
  onResultUpdate: onAnswerUpdate,
  onDone,
}) => {
  const inputRef = useRef<HTMLInputElement>(null);

  const [query, setQuery] = useState<string>("");
  const [apiKey, setApiKey] = useState<string>("");
  const [loading, setLoading] = useState<boolean>(false);
  const [error, setError] = useState<boolean>(false);

  useEffect(() => {
    const API_KEY = localStorage.getItem(LOCAL_KEY_ID);
    if (API_KEY) {
      setApiKey(JSON.parse(API_KEY));
    }
  }, []);

  const handleSearch = async () => {
    if (!query) {
      setError(true);
      return;
    }

    setLoading(true);
    const sources = await fetchSources();
    await handleStream(sources);
  };

  const fetchSources = async () => {
    const response = await fetch("/api/sources", {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
      },
      body: JSON.stringify({ query }),
    });

    if (!response.ok) {
      setLoading(false);
      throw new Error(response.statusText);
    }

    const { sources }: { sources: Source[] } = await response.json();
    return sources;
  };

  const handleStream = async (sources: Source[]) => {
    if (sources.length <= 0) {
      setLoading(false);
      onAnswerUpdate(UNRELATED_ANSWER);
      onSearch({ query, sourceLinks: []})
      return;
    }
    try {
      const prompt = endent`Provide a 1 to 3 sentence answer (with the last sentence ending with a full stop) to the query based on the following sources. Be original, concise, accurate, and helpful. Cite sources as [1] or [2] or [3] after each sentence (not just the very end) to back up your answer (Ex: Correct: [1], Correct: [2][3], Incorrect: [1, 2]).
      ${sources
        .map((source, idx) => `Source [${idx + 1}]:\n${source.text}`)
        .join("\n\n")}
      `;

      const response = await fetch("/api/answer", {
        method: "POST",
        headers: {
          "Content-Type": "application/json",
        },
        body: JSON.stringify({ prompt, apiKey }),
      });

      if (!response.ok) {
        setLoading(false);
        throw new Error(response.statusText);
      }

      setLoading(false);
      onSearch({ query, sourceLinks: sources.map((source) => source.url) });

      const data = response.body;

      if (!data) {
        return;
      }

      const reader = data.getReader();
      const decoder = new TextDecoder();
      let done = false;

      while (!done) {
        const { value, done: doneReading } = await reader.read();
        done = doneReading;
        const chunkValue = decoder.decode(value);
        onAnswerUpdate(chunkValue);
      }

      onDone(true);
    } catch (err) {
      onAnswerUpdate("Error");
    }
  };

  const handleKeyDown = (e: KeyboardEvent<HTMLInputElement>) => {
    if (e.key === "Enter") {
      handleSearch();
    }
  };

  return (
    <>
      {loading ? (
        <>
          <span className='inline-flex'>
            <div className='h-6 w-6 animate-spin rounded-full border-2 border-solid border-current border-r-transparent align-[-0.125em] motion-reduce:animate-[spin_1.5s_linear_infinite] mr-2'></div>
            <p>Thinking...</p>
          </span>
        </>
      ) : (
        <>
          {apiKey.length === 51 || apiKey === "local" ? (
            <div className='relative w-full'>
              <Input
                id="search-input"
                label="Enter a health related query"
                name="search"
                hint="e.g What is an HRT PPC?"
                placeholder='type something here'
                ref={inputRef}
                value={query}
                type="text"
                onChange={(e: ChangeEvent<HTMLInputElement>) => setQuery(e.target.value)}
                onKeyDown={handleKeyDown}
                error={error ? 'must provide a search term' : ''}
              />
              <Button id='submit' as='a' data-prevent-double-click onClick={handleSearch}>
                Submit
              </Button>
            </div>
          ) : (
            <InsetText visuallyHiddenText='An Open AI Key must be set in order to search.'>
              <p>An Open AI Key must be set in order to search.</p>
            </InsetText>
          )}
          <ActionLink href="/settings">You can change the OpenAI API Key here</ActionLink>
        </>
      )}
    </>
  );
};
