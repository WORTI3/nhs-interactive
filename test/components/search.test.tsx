import { render, fireEvent, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Search } from '../../app/[locale]/components/Search';
import { NextIntlClientProvider } from 'next-intl';
import { LOCALES } from '../../app/utils/constants';
import { act } from 'react-dom/test-utils';

describe('Search', () => {
  function toRender(messages: any, locale: string) { 
    return (<NextIntlClientProvider messages={messages} locale={locale}>
      <Search
        onSearch={jest.fn()}
        onResultUpdate={jest.fn()}
        onDone={jest.fn()}
      />
    </NextIntlClientProvider>)
  };

  test.each(LOCALES)(
    'rendered component should match snapshot (%s)',
    (locale) => {
      const messages = require(`../../locales/${locale}.json`);
      const { container } = render(toRender(messages, locale));
      expect(container).toMatchSnapshot();
    }
  );

  test.each(LOCALES)(
    "displays error message when submit with empty query (%s)",
    async (locale) => {
      const messages = require(`../../locales/${locale}.json`);
      render(toRender(messages, locale));
      const submitButton = await screen.findByText(messages.search.button.submit);
      act(() => {
      fireEvent.click(submitButton);
      });
      const errorMessage = await screen.findByText(messages.search.errors.too_small);
      expect(errorMessage).toBeInTheDocument();
    }
  );

  test.each(LOCALES)(
    "displays loading spinner when searching",
    async (locale) => {
      const messages = require(`../../locales/${locale}.json`);
      const { container } = render(toRender(messages, locale));
      const submitButton = await screen.findByText(messages.search.button.submit);
      
      act(() => {
        fireEvent.change(screen.getByRole("textbox"), {
          target: { value: "what is an HRT PPC" },
        });
      });
      act(() => {
      fireEvent.click(submitButton);
      });
      const loadingSpinner = await screen.findByTestId("animated-progress");
      // expect(loadingSpinner).toBeInTheDocument(); fix test
      expect(container.firstChild).toMatchSnapshot();
    }
  );

  test.each(LOCALES)('displays input label text', (locale) => {
    const messages = require(`../../locales/${locale}.json`);
    render(toRender(messages, locale));
    const inputLabel = screen.getByLabelText(messages.search.query);
    expect(inputLabel).toBeInTheDocument();
  });
});
