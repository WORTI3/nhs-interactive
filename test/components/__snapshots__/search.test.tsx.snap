// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search displays loading spinner when searching 1`] = `
<div
  class="relative w-full"
>
  <div
    class="nhsuk-character-count nhsuk-form-group nhsuk-form-group--error"
  >
    <label
      class="nhsuk-label"
      for="search-input"
    >
      Enter a health related query
    </label>
    <span
      class="nhsuk-hint"
    >
      e.g What is an HRT prescription?
    </span>
    <span
      class="nhsuk-error-message"
    >
      the search query must be more than 4 letters
    </span>
    <textarea
      class="nhsuk-textarea nhsuk-js-character-count nhsuk-textarea--error"
      id="search-input"
      maxlength="100"
      name="search"
      placeholder="type something here"
      rows="1"
    />
  </div>
  <button
    class="nhsuk-button    disabled"
    data-module="nhsuk-button"
    data-prevent-double-click="false"
    disabled=""
    id="submit"
    type="button"
  >
    Submit
  </button>
  <span
    class="nhsuk-body-s"
  >
    Queries and results are stored for further analysis. We wont pass personally identifiable information to OpenAI or our internal systems.
  </span>
</div>
`;

exports[`Search displays loading spinner when searching 2`] = `
<div
  class="relative w-full"
>
  <div
    class="nhsuk-character-count nhsuk-form-group nhsuk-form-group--error"
  >
    <label
      class="nhsuk-label"
      for="search-input"
    >
      Rhowch gwestiwn sy'n ymwneud â iechyd
    </label>
    <span
      class="nhsuk-hint"
    >
      e.e. Beth yw HRT PPC?
    </span>
    <span
      class="nhsuk-error-message"
    >
      Rhaid i'r ymholiad chwilio fod yn fwy na 4 llythyren
    </span>
    <textarea
      class="nhsuk-textarea nhsuk-js-character-count nhsuk-textarea--error"
      id="search-input"
      maxlength="100"
      name="search"
      placeholder="teipiwch rhywbeth yma"
      rows="1"
    />
  </div>
  <button
    class="nhsuk-button    disabled"
    data-module="nhsuk-button"
    data-prevent-double-click="false"
    disabled=""
    id="submit"
    type="button"
  >
    Cyflwyno
  </button>
  <span
    class="nhsuk-body-s"
  >
    Cedwir ymholiadau a chanlyniadau ar gyfer dadansoddiad pellach. Ni fyddwn yn trosglwyddo gwybodaeth bersonol adnabyddadwy i OpenAI neu ein systemau mewnol.
  </span>
</div>
`;

exports[`Search displays loading spinner when searching 3`] = `
<div
  class="relative w-full"
>
  <div
    class="nhsuk-character-count nhsuk-form-group nhsuk-form-group--error"
  >
    <label
      class="nhsuk-label"
      for="search-input"
    >
      Cuir fiosrachadh a bhaineas ri slàinte
    </label>
    <span
      class="nhsuk-hint"
    >
      e.g. Dè tha PPC HRT?
    </span>
    <span
      class="nhsuk-error-message"
    >
      feumaidh barrachd air 4 litrichean a bhith anns a' cheist rannsachaidh
    </span>
    <textarea
      class="nhsuk-textarea nhsuk-js-character-count nhsuk-textarea--error"
      id="search-input"
      maxlength="100"
      name="search"
      placeholder="cuir rudeigin an seo"
      rows="1"
    />
  </div>
  <button
    class="nhsuk-button    disabled"
    data-module="nhsuk-button"
    data-prevent-double-click="false"
    disabled=""
    id="submit"
    type="button"
  >
    Cuir ris
  </button>
  <span
    class="nhsuk-body-s"
  >
    Tha ceistean agus toraidhean air an stòradh airson tuilleadh sgrùdaidh. Cha toir sinn seachad fiosrachadh a dh' aithnicheas sinn gu pearsanta gu OpenAI no na siostaman a-staigh againn.
  </span>
</div>
`;

exports[`Search rendered component should match snapshot (cy) 1`] = `
<div>
  <div
    class="relative w-full"
  >
    <div
      class="nhsuk-character-count nhsuk-form-group "
    >
      <label
        class="nhsuk-label"
        for="search-input"
      >
        Rhowch gwestiwn sy'n ymwneud â iechyd
      </label>
      <span
        class="nhsuk-hint"
      >
        e.e. Beth yw HRT PPC?
      </span>
      <textarea
        class="nhsuk-textarea nhsuk-js-character-count "
        id="search-input"
        maxlength="100"
        name="search"
        placeholder="teipiwch rhywbeth yma"
        rows="1"
      />
    </div>
    <button
      class="nhsuk-button    "
      data-module="nhsuk-button"
      data-prevent-double-click="false"
      id="submit"
      type="button"
    >
      Cyflwyno
    </button>
    <span
      class="nhsuk-body-s"
    >
      Cedwir ymholiadau a chanlyniadau ar gyfer dadansoddiad pellach. Ni fyddwn yn trosglwyddo gwybodaeth bersonol adnabyddadwy i OpenAI neu ein systemau mewnol.
    </span>
  </div>
</div>
`;

exports[`Search rendered component should match snapshot (en) 1`] = `
<div>
  <div
    class="relative w-full"
  >
    <div
      class="nhsuk-character-count nhsuk-form-group "
    >
      <label
        class="nhsuk-label"
        for="search-input"
      >
        Enter a health related query
      </label>
      <span
        class="nhsuk-hint"
      >
        e.g What is an HRT prescription?
      </span>
      <textarea
        class="nhsuk-textarea nhsuk-js-character-count "
        id="search-input"
        maxlength="100"
        name="search"
        placeholder="type something here"
        rows="1"
      />
    </div>
    <button
      class="nhsuk-button    "
      data-module="nhsuk-button"
      data-prevent-double-click="false"
      id="submit"
      type="button"
    >
      Submit
    </button>
    <span
      class="nhsuk-body-s"
    >
      Queries and results are stored for further analysis. We wont pass personally identifiable information to OpenAI or our internal systems.
    </span>
  </div>
</div>
`;

exports[`Search rendered component should match snapshot (gd) 1`] = `
<div>
  <div
    class="relative w-full"
  >
    <div
      class="nhsuk-character-count nhsuk-form-group "
    >
      <label
        class="nhsuk-label"
        for="search-input"
      >
        Cuir fiosrachadh a bhaineas ri slàinte
      </label>
      <span
        class="nhsuk-hint"
      >
        e.g. Dè tha PPC HRT?
      </span>
      <textarea
        class="nhsuk-textarea nhsuk-js-character-count "
        id="search-input"
        maxlength="100"
        name="search"
        placeholder="cuir rudeigin an seo"
        rows="1"
      />
    </div>
    <button
      class="nhsuk-button    "
      data-module="nhsuk-button"
      data-prevent-double-click="false"
      id="submit"
      type="button"
    >
      Cuir ris
    </button>
    <span
      class="nhsuk-body-s"
    >
      Tha ceistean agus toraidhean air an stòradh airson tuilleadh sgrùdaidh. Cha toir sinn seachad fiosrachadh a dh' aithnicheas sinn gu pearsanta gu OpenAI no na siostaman a-staigh againn.
    </span>
  </div>
</div>
`;
